### PR TITLE
Fix #6651: In x86 anal, immediate -> val not ptr

### LINKED ISSUE #6651: analysis puts value in wrong field,  $v / $m fail
https://github.com/radare/radare2-regressions/pull/690

https://github.com/radare/radare2book/pull/44/commits/73941c3c9425cf0f042e35e794bfcd947d091dea

### PR DESCRIPTION
**Pull request by SchumBlubBlub**
closes #6651 

**Update:**
After this, merge
https://github.com/radare/radare2-regressions/pull/690
https://github.com/radare/radare2-regressions/pull/693

### CODE REVIEW

**Comment by radare** [comment]
many tests are broken with this change because some analysis expect the pointer to be in ptr not in val. maybe we should set in both?

**Comment by SchumBlubBlub** [comment]
I'm working on fixing those. may be good idea to have new variable (`$vm`?) which is "whichever apply". Sound ok?

**Comment by SchumBlubBlub** [comment]
All fixed.
https://github.com/radare/radare2-regressions/pull/690
https://github.com/radare/radare2-regressions/pull/693

everything green.

**Comment by radare** [comment]
please squash and rebase

File: libr/anal/fcn.c [hunk 1]
```diff
@@ -901,8 +910,8 @@ static int fcn_recurse(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut8 *buf, ut6
 			break;
 		case R_ANAL_OP_TYPE_CJMP:
 			if (anal->opt.cjmpref) {
-				(void) r_anal_fcn_xref_add (anal, fcn,
-					op.addr, op.jump, R_ANAL_REF_TYPE_CODE);
+				(void)r_anal_fcn_xref_add (anal, fcn,
+							op.addr, op.jump, R_ANAL_REF_TYPE_CODE);
```
Comment (radare) [inline libr/anal/fcn.c]: autoindent fail

File: libr/anal/fcn.c [hunk 2]
```diff
@@ -1286,15 +1293,15 @@ R_API int r_anal_fcn_add(RAnal *a, ut64 addr, ut64 size, const char *name, int t
...
+	sdb_set (DB, sdb_fmt (0, "fcn.0x%08" PFMT64x, addr), "TODO", 0); // TODO: add more info here
 #endif
 	return append? r_anal_fcn_insert (a, fcn): true;
 }
 
 R_API int r_anal_fcn_del_locs(RAnal *anal, ut64 addr) {
 	RListIter *iter, *iter2;
 	RAnalFunction *fcn, *f = r_anal_get_fcn_in (anal, addr,
-		R_ANAL_FCN_TYPE_ROOT);
+						R_ANAL_FCN_TYPE_ROOT);
```
Comment (radare) [inline libr/anal/fcn.c]: autoindent fail

File: libr/core/disasm.c [hunk 3]
```diff
@@ -391,52 +390,82 @@ static void ds_print_spacy(RDisasmState *ds, int pre) {
...
-	ds->color_num = P(num): Color_CYAN;
-	ds->color_mov = P(mov): Color_WHITE;
-	ds->color_invalid = P(invalid): Color_BRED;
-	ds->color_gui_cflow = P(gui_cflow): Color_YELLOW;
-	ds->color_gui_dataoffset = P(gui_dataoffset): Color_YELLOW;
-	ds->color_gui_background = P(gui_background): Color_BLACK;
-	ds->color_gui_alt_background = P(gui_alt_background): Color_GRAY;
-	ds->color_gui_border = P(gui_border): Color_BGGRAY;
+#define P(x) (core->cons && core->cons->pal.x)? core->cons->pal.x
+	ds->color_comment = P (comment)
```
Comment (radare) [inline libr/core/disasm.c]: UNREADABLE

File: libr/core/disasm.c [hunk 4]
```diff
@@ -595,19 +624,19 @@ static void ds_reflines_init(RDisasmState *ds) {
 	if (ds->show_lines) {
 		ds_reflines_fini (ds);
 		anal->reflines = r_anal_reflines_get (anal,
-			ds->addr, ds->buf, ds->len, ds->l,
-			ds->linesout, ds->show_lines_call);
+						ds->addr, ds->buf, ds->len, ds->l,
```
Comment (radare) [inline libr/core/disasm.c]: bad autoindent

File: libr/core/disasm.c [hunk 5]
```diff
@@ -873,13 +902,13 @@ static void ds_show_refs(RDisasmState *ds) {
 			r_core_read_at (ds->core, ref->at, buf, sizeof (buf));
 			r_anal_op (ds->core->anal, &aop, ref->at, buf, sizeof (buf));
 			if ((aop.type & R_ANAL_OP_TYPE_MASK) == R_ANAL_OP_TYPE_UCALL) {
-				RAnalFunction * fcn = r_anal_get_fcn_at (ds->core->anal,
-					ref->addr, R_ANAL_FCN_TYPE_NULL);
-				_ds_comment_align_ (ds, true, false);
+				RAnalFunction *fcn = r_anal_get_fcn_at (ds->core->anal,
+									ref->addr, R_ANAL_FCN_TYPE_NULL);
```
Comment (radare) [inline libr/core/disasm.c]: more bad autoindent

File: libr/core/disasm.c [hunk 6]
```diff
@@ -1750,10 +1777,10 @@ static void ds_print_cycles(RDisasmState *ds) {
 static void ds_print_stackptr(RDisasmState *ds) {
 	if (ds->show_stackptr) {
 		r_cons_printf ("%5d%s", ds->stackptr,
-			ds->analop.type == R_ANAL_OP_TYPE_CALL?">":
-			ds->analop.stackop == R_ANAL_STACK_ALIGN? "=":
-			ds->stackptr > ds->ostackptr? "+":
-			ds->stackptr < ds->ostackptr? "-": " ");
+			ds->analop.type == R_ANAL_OP_TYPE_CALL? ">" :
+									ds->analop.stackop == R_ANAL_STACK_ALIGN? "=" :
```
Comment (radare) [inline libr/core/disasm.c]: horrible autoindent

File: libr/core/disasm.c [hunk 7]
```diff
@@ -1867,11 +1894,10 @@ static void ds_print_trace(RDisasmState *ds) {
 static void ds_adistrick_comments(RDisasmState *ds) {
 	if (ds->adistrick) {
 		ds->middle = r_anal_reflines_middle (ds->core->anal,
-			ds->core->anal->reflines, ds->at, ds->analop.size);
+						ds->core->anal->reflines, ds->at, ds->analop.size);
```
Comment (radare) [inline libr/core/disasm.c]: more bad autoindent

File: libr/core/disasm.c [hunk 8]
```diff
@@ -3096,166 +3128,166 @@ static void ds_print_esil_anal(RDisasmState *ds) {
...
+				const char *cc;
+				nargs = r_anal_type_func_args_count (core->anal, key);
+				// HACK: remove other comments
+				delete_last_comment (ds);
+				if (ds->show_color) {
+					ds_comment_esil (ds, true, false, ds->pal_comment);
+				}
+				ds_align_comment (ds);
+				ds_comment_esil (ds, ds->show_color? false: true, false,
+						"; %s%s%s(", r_str_get (fcn_type), (fcn_type && *fcn_type &&
```
Comment (radare) [inline libr/core/disasm.c]: horrible autoindent

File: libr/anal/fcn.c (hunk 1 above)
Comment (SchumBlubBlub) [inline libr/anal/fcn.c]: that's what sys/indent does. here and everywhere else.

**Comment by radare** [comment]
i know, thats why there’s a whitelist and it needs to be enhanced, do your uncrustify does that right?

**Comment by SchumBlubBlub** [comment]
I haven't tried and it's not "my uncrustify", I was trying to help.

**Comment by SchumBlubBlub** [comment]
I'm just going to step away here, for a day or two... cool down.

**Comment by radare** [comment]
ok i have tested your uncrustify and it keeps having the same issues.

* cant enforce braces on oneliner blocks
* cant enforce oneliner function declarations
* onelinered if blocks are not expanded
* arguments in a long line get a lot of indent
* .. 

and i have just tested it in 2 files. 

i have packaged it into r2pm

btw, the idea behind the autoindentation is to cleanup the code to make it consumable by the formatter without destroying too much, this sometimes makes the code better, and sometimes worst, so it needs some manual review. we should try to enforce indent.sh you are right on that, but this script is still not perfect

**Comment by SchumBlubBlub** [comment]
morning. Removed the sys/indent fixes commit. I think i fixed this manually already, and should be mergable together with r2r PR.

**Comment by radare** [comment]
i have updated the sys/indent.sh to support uncrustify too. also , tweaked a bit the uncrustify.cfg .. but its still not correct, but seems to be btter than last time i tried. i’ll try to improve on that aspect and hopefully before 1.4 get all r2 code base indented automatically.

**Comment by radare** [comment]
but better keep that PR without indentation stuff to avoid confusion or unnecessary rebases

**Comment by SchumBlubBlub** [comment]
Thank you! I pushed an updated uncrustify.cfg a few hours ago to fix *some* of your comments, see if it is better?

**Comment by radare** [comment]
i’ll write a document explaining the correct process to autoindent r2 and what can people do to improve the situation

thanks!

about this pr.. i want to do some stuff before merging, because this will break some analysis that follows ptr, when imm == ptr, but i have no time to check that now

**Comment by SchumBlubBlub** [comment]
np, but I fix all the r2r that broke, the prs is there..

**Comment by radare** [comment]
fix = reviewed all the cases and determined the test was wrong?

**Comment by SchumBlubBlub** [comment]
Yes, not just "change the EXPECT", I looked why.

**Comment by radare** [comment]
Rebased and squashed here https://github.com/radare/radare2/pull/6969

**Pull request closed without merging**